dotnet/src/SemanticKernel.Core/PromptTemplate/NoopPromptTemplateFactory.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Provides an implementation of <see cref="IPromptTemplateFactory"/> which creates no operation instances of <see cref="IPromptTemplate"/>.
+/// </summary>
+public sealed class NoopPromptTemplateFactory : IPromptTemplateFactory
```

Comment:
As there's no instance state, we could expose a singleton via a static property.